### PR TITLE
Fix #406 - Settings for notify by a given level

### LINKED ISSUE #406: Option to only show notifications on test failures (successful runs silent)
I think it would be great to be able to disable the notifications when the tests succeeds.

### PR DESCRIPTION
**Pull request by ruzz311**
Added configuration to notifications based on test result status as was discussed in issue #406 

Screen-capture Preview:
goconvey-notification-settings

### CODE REVIEW

File: web/client/index.html [hunk 1]
```diff
@@ -483,5 +497,21 @@
...
+		<script src="/resources/js/lib/jquery.js"></script>
+		<script src="/resources/js/lib/jquery-ui.js"></script>
+		<script src="/resources/js/lib/jquery.pretty-text-diff.js"></script>
+		<script src="/resources/js/lib/jquery.tipsy.js"></script>
+
+		<!-- Script ordering is important -->
+		<script src="/resources/js/poller.js"></script>
+		<script src="/resources/js/convey.js"></script>
+		<script src="/resources/js/config.js"></script>
+		<script src="/resources/js/goconvey.js"></script>
```
Comment (riannucci) [inline web/client/index.html]: can we do this move as a separate change?

File: web/client/resources/js/goconvey.js [hunk 2]
```diff
@@ -477,8 +498,16 @@ function loadSettingsFromStorage()
 	convey.uiEffects = uiEffects === "true";
 	enumSel("ui-effects", uiEffects);
 
-	if (notif())
-		$('#toggle-notif').toggleClass("fa-bell-o fa-bell " + convey.layout.selClass);
+	enumSel("notification", notif() ? 'on' : 'off');
```
Comment (riannucci) [inline web/client/resources/js/goconvey.js]: would it be possible to make this actually a boolean? The 'on' and 'off' strings everywhere give me the heebie jeebies.

File: web/client/resources/js/goconvey.js [hunk 3]
```diff
@@ -769,7 +798,9 @@ function process(data, status, jqxhr)
 	convey.intervalFuncs.momentjs();
 
 	// Show notification, if enabled
-	if (notif())
+	var levelRegex = new RegExp("("+convey.notificationLevel+")", "i");
+	var levelEnabled = current().overall.status.class.match(levelRegex);
```
Comment (riannucci) [inline web/client/resources/js/goconvey.js]: isn't this a bit overkill? This implies that data-notification-level is actually a regex fragment.

If this is a good approach, then why have 'all' and not just '.*' ?

**Comment by riannucci** [comment]
This is really cool! I'm not an html/js guru though, so hopefully someone else who is can take a look. Other than that :+1: .

File: web/client/resources/js/goconvey.js (hunk 3 above)
Comment (riannucci) [inline web/client/resources/js/goconvey.js]: Yeah I think a hash might be a decent compromise. Regexs are a bit of an orbital nuke here I think :D

File: web/client/resources/js/goconvey.js (hunk 2 above)
Comment (ruzz311) [inline web/client/resources/js/goconvey.js]: Fixed: https://github.com/smartystreets/goconvey/pull/418/files#diff-02a2830d03f0667451f10d94a0a6c1aeR501

File: web/client/resources/js/goconvey.js (hunk 3 above)
Comment (ruzz311) [inline web/client/resources/js/goconvey.js]: I guess I just like the idea of being able to add a new option or a more specific filtering of goconvey's various states without having to dig through 1.35k lines of javascript to add additional logic. For example, I get the benefit of a build failure and a test failure by using the regex string method.

**Comment by ruzz311** [comment]
@riannucci @mholt @mdwhatcott - This has been sitting for way too long.  I just rebased and pushed to resolve some conflicts.  I'm planning on merging in the next day unless I see any objections in the next day or so.

**Comment by mdwhatcott** [comment]
Looks good to me.

**Comment by ruzz311** [comment]
Thanks, just trying to clean up a little.

**Comment by toftware** [comment]
Thank you for this.

**Pull request merged**